Augur 2333 parameterize connectors

justinbarry opened the pull request:
Things to note:
1. The sort param in load-markets is commented out. Sorting in the getters currently causes and exception to be thrown.
2. The ethnode url is hardcoded on the client side. The UX will need to be sorted out for this. ( paging @bthaile )
3. Google bot language level transpiring has been removed in webpack for external dependencies. Unknown is this impacts indexibility (A new version of the google crawler was just announced at google i/o).
closes #2333

justinbarry committed:
Export connectors from top level of SDK

justinbarry committed:
Strongly type pouchDB config object

justinbarry committed:
Add networks.json config file to augur artifacts

justinbarry committed:
Spread args out to match behaviour of parent class

justinbarry committed:
Parameterize connector with ethNodeUrl and account

brianosaurus commented on packages/augur-sdk/src/connector/ww-connector.ts:
```diff
@@ -1,35 +1,14 @@
-import { UploadBlockNumbers, Addresses } from "@augurproject/artifacts";
-
-const settings = require("@augurproject/sdk/src/state/settings.json");
+import {API} from "../state/getter/API";
```
I think your editor is auto formatting for no spaces in the import lists but others (mine included) are doing the opposite. We should sort that out some day.

brianosaurus commented on packages/augur-sdk/src/events.ts:
```diff
@@ -11,7 +11,7 @@ class EventNameEmitter extends EventEmitter {
       );
     });
-    return super.emit(type, eventsWithName);
+    return super.emit(type, ...eventsWithName);
```
Curious why this was needed? I believe it'll break the connectors as written. They map over the event data but that's probably a bug. We'll need to retool the connectors to allow more than one argument to a Callback type. Currently the callback type is: 
```ts
export type Callback = (data: any) => void;
```

justinbarry replied:
I added it because it is inline with the type definition of the parent's method definition. 

See: https://nodejs.org/api/events.html#events_emitter_emit_eventname_args

brianosaurus commented on packages/augur-sdk/src/state/index.ts:
```diff
@@ -1,48 +1 @@
-//     console.log(e);
-//   }
-// })();
+const settings = require("./settings.json");
```
I think this needs to be 
```ts
@augurproject/sdk/src/state/settings.json
```

justinbarry replied:
Adding the file to the include block in tsconfig.json did the trick. Relative json file imports work again!

justinbarry committed:
Kick off WebWorker sync from main  thread

justinbarry committed:
UI tsconfig inherits base lib config

justinbarry committed:
Add worker loader to webpack common config

justinbarry committed:
Add missing .json files to tsconfig "include" block

The compiler will not copy them to the build dir unless explicitly
referenced here.

justinbarry committed:
Move SDK web worker to UI package because webpack

brianosaurus committed:
Fixes connector initialization issue

brianosaurus committed:
removing console.log

justinbarry committed:
Use account address provided by metamask when logged in

justinbarry committed:
Fix augur-node NPE on server start

justinbarry committed:
Explicitly include json config file in UI TS build

justinbarry committed:
Fix "end of json input" error

justinbarry committed:
Add missing argument causing npe

justinbarry committed:
Remove errant double promise wrapper of getter result

Return type altered as follows:
Promise<Promise<Array<address>>> -> Promise<Array<address>>

justinbarry committed:
Refer to DB as promise to prevent async init issue

justinbarry committed:
Add source-map-loader to UI webpack build

justinbarry committed:
Load market ids from the webworker

justinbarry committed:
Remove ts-ignore as babel ignores it anyway

justinbarry committed:
Separate out db creation from controller

We need a "readonly"  version of the DBs in the webworker connector

pgebheim commented on packages/augur-node/src/runServer.ts:
```diff
@@ -9,14 +9,19 @@ import { JsonRpcProvider } from "ethers/providers/json-rpc-provider";
 
 import { Addresses } from "@augurproject/artifacts";
 import { ContractDependenciesEthers } from "contract-dependencies-ethers";
+import {EmptyConnector} from "@augurproject/sdk/build";
```
importing from `build` Perhaps 'cause this is just to get AN working?

justinbarry replied:
Nope. Just bad auto complete! Good catch!

justinbarry committed:
Update tests to use DB promise vs db ref itself

justinbarry committed:
Merge branch 'master' into Augur-2333-parameterize-connectors
